src/core/server/config/env.mock.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Env } from './env';
+
+export function createMockEnv(options: { dist?: boolean } = {}): Env {
+  return {
+    // required by CspConfig
+    packageInfo: {
+      dist: options.dist ?? true,
+    },
+  } as any;
+}
```

Comment:
We should really try to remove the hardcoded pkg access in 

https://github.com/elastic/kibana/blob/24542c3ad747ff6bb024e315e5475fb6cdb04a77/src/core/server/config/env.ts#L25

and pass it as a constructor parameter as we do with EnvOptions. It would allow to create concrete `Env` instance for tests way more easily. The current approach of mocking the package is tedious.

This will have to do for now, even if having a type matching the actual class properties would be better. This implementation would break anything trying to access something else that packageInfo.dist, and our mocks are supposed to be re-usable.

created https://github.com/elastic/kibana/issues/55470